### ISSUE TITLE
Talkback

### ISSUE THREAD

**Issue by mottosso** [pyblish/pyblish-qml#95]
### Goal

Simplify codebase and enable remote testing.

### Motivation

To get the GUI to show from within a host, the GUI must be told to do that somehow. Currently, communicating is one-way from GUI to host. The GUI can tell the host to do things, like publish, the but host can't tell the GUI to do anything. To work around this fact, the GUI sends a long-running request to the host, basically asking "Did you want something?" and not withdrawing from a response until a long time (it actually asks once every second, to which the host typically replies "No, not really"). When the host then finally says "Hey GUI, I'd like you to show now", the GUI can respond by finally showing.

In short, it's a hack.

### Implementation

Instead, let the GUI provide it's own endpoint for which external communication may be had.

```python
class QmlService(object):
    def __init__(self, app):
        self.app = app

    def show(self, port):
        """Show the GUI"""
        self.app.show_signal.emit(port)
        return True

def _server(port, service):
    server = SimpleXMLRPCServer(
        ("127.0.0.1", port),
        allow_none=True,
        logRequests=False)

    server.register_introspection_functions()
    server.register_instance(service)

    return server
```

This enabled control of the GUI from anywhere, including from hosts but also tests.

**Comment by mottosso** [comment]
Implemented in 0.2.10
